frontend/app/components/collapsible.tsx

```diff
@@ -0,0 +1,48 @@
+import type { ComponentProps } from 'react';
+
+import { cn } from '~/utils/tw-utils';
+
+const variants = {
+  default: 'border border-gray-300 bg-gray-200 text-slate-700 hover:bg-neutral-300',
+};
+
+export interface CollapsableDetailsProps extends ComponentProps<'details'> {
+  id: string;
+  title: string;
+  required?: boolean;
+  label?: string;
+  variant?: keyof typeof variants;
+}
```

Comment:
You make `id` as a required prop and add `summary` as a prop to provide the summary's props.

```suggestion
export interface CollapsableDetailsProps extends ComponentProps<'details'> {
  id: string;
  summary: Omit<ComponentProps<typeof CollapsableSummary>, 'id'>;
}
```

Reply:
Wouldn't the header title be needed?

Reply:
Yeah, and it's `summary` prop. I refactored my suggestion and set `summary` type to `ReactNode`.
https://dev.to/elhamnajeebullah/react-typescript-what-is-reactnode-and-when-to-use-it-3660

Reply:
Fixed: ead183074aeaba144439cef11bace94d8046263c